### ISSUE TITLE
Feature request: respect default search engine in vomnibar

### ISSUE THREAD

**Issue by Vinnl** [philc/vimium#2598]
(I'm one of the VimFx refugees, so apologies if this request is too Firefox-specific.)

When I search with the Vomnibar, that search is executed in Google. My default search engine (in Firefox, but I believe you can set on in Chrome too?) is not Google, however. It would be nice if Vimium could find out which search engine I've set as default and use that for searches.

**Comment by mrmr1993** [comment]
Neither Chrome nor Firefox provide a WebExtensions API for us to access the browser's search engines, so you will have to change it manually if you want it to match the default.

You can change the default and/or available engines by following [the steps here](https://github.com/philc/vimium/wiki/Tips-and-Tricks#keyboard-shortcuts-for-custom-search-engines).

**Comment by leoschwarz** [comment]
For Firefox there is some initial upstream discussion about providing such an API: https://bugzilla.mozilla.org/show_bug.cgi?id=1352598

@mrmr1993 The linked docs actually don't explain how to change the default search engine, only how to add custom search engines. I had to click on "Show Advanced Options" and then copy the search prefix url into the default search engine field.

**Comment by jorgesumle** [comment]
If this cannot be solved now, I propose changing the Vimiums's search engine to something more privacy-respecting like Brave Search or DuckDuckGo. I also think that reviewing and accepting https://github.com/philc/vimium/pull/3693 would be a great improvement. I had to search a lot to change this, because it's in the advanced options, and I didn't know how to get there.

**Comment by ayn2op** [comment]
- https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/search
- https://developer.chrome.com/docs/extensions/reference/search

**Comment by dudicoco** [comment]
> If this cannot be solved now, I propose changing the Vimiums's search engine to something more privacy-respecting like Brave Search or DuckDuckGo. I also think that reviewing and accepting #3693 would be a great improvement. I had to search a lot to change this, because it's in the advanced options, and I didn't know how to get there.

@jorgesumle how do you know that these are privacy respecting without access to their backends?

**Comment by jorgesumle** [comment]
> @jorgesumle how do you know that these are privacy respecting without access to their backends?

Their business model isn't based on collecting lots of user data, they don't place tracking shit like this `?q=asda&sca_esv=941af24518045317&source=hp&ei=BC_zZYT5CZ-8xc8Pp9yx6AQ&iflsig=ANes7DEAAAAAZfM9FNwznii-d8S-bRQ9Dr6MosVSrS46&ved=0ahUKEwiEi6WBn_SEAxUfXvEDHSduDE0Q4dUDCA8&uact=5&oq=asda&gs_lp=Egdnd3Mtd2l6IgRhc2RhSAtQAFgAcAB4AJABAJgBAKABAKoBALgBA8gBAJgCAKACAJgDAJIHAKAHAA&sclient=gws-wiz` when you make a query, and they claim to be privacy-respecting. I obviously don't know what happens in their backend, but I'm 100 % sure they are more privacy friendly than Google.

**Comment by philc** [comment]
1. I think the UI for changing the default search engine is now less of an issue (see my comment on #3693).
2. Vimium doesn't have any particular opinion on search engines. If we have to pick one, then to reduce user friction, we should use the most popular one (which is Google, by far) to reduce the proportion of people that get surprised when they search in the Vomnibar for the first time, and who then have to dig around to change the search engine setting.
3. If the user cares about using a specific search engine, presumably they've already changed it in their browser settings, so #4330 is a great solution to this problem. I wasn't aware that there was a search API that extensions could use. It was introduced in Chrome 87. I think that's the way forward.

**Comment by kungfubeaner** [comment]
I honestly thing being able to set the default search enging was a better option and more customizable rather than default.  I was using this as if i search from vomnibar it would search brave, and firefoxes url bar google as a choice but oh well...

**Comment by habina** [comment]
Please bring the customized default search setting back to Vimiuim option. thanks.

**Comment by philg-dev** [comment]
In https://github.com/philc/vimium/issues/4669 we've found a fairly annoying anti-feature. For more details see the full discussion over there.

TL;DR:
Search engines that are defined to use `HTTP POST` in their [OpenSearch.xml](https://developer.mozilla.org/en-US/docs/Web/XML/Guides/OpenSearch) will run into CSP issues, when triggered from a webextension with `search.search()` or `search.query()` due to the `Origin` header being set to the extension's URL. When using the native browser URL bar the `POST` request works fine.

In the example it was regarding the DuckDuckGo HTML and Lite variants, which are explicitly "non-JS sites". Despite their OpenSearch descriptors specifying `POST`, the server also allows `GET` requests. This makes it possible to simply add those search engines as custom search engines to Vimium, since Vimium always uses `GET` requests for those. 

```
ddg: https://html.duckduckgo.com/html?q=%s DuckDuckGo HTML
```

Unfortunately that would require the user to use a prefix all the time. If we got the option back to specify an Vimium default search engine that can be different from the browser's default search engine, this workaround could help people that are affected by this limitation.
